docs/ref/building.rst

```diff
@@ -5,15 +5,85 @@ Configuring, Building, & Using
 .. default-domain:: std
 .. default-role:: any
 
-Building |amongoc| is supported with the following build tools:
+Building |amongoc| requires a C++20 compiler. The following tools are known to
+work:
 
 - CMake_ 3.25 or newer
-- GCC ≥12.0 **or** Clang ≥17.0
+- GCC ≥12.0\ [#fn-redhat-issue]_ **or** Clang ≥17.0
 - Earthly_ 0.8 or newer (for :ref:`building with Earthly <building.earthly>`)
 
+Building on Windows or with MSVC is not currently supported.
+
 .. _CMake: https://cmake.org/
 .. _Earthly: https://earthly.dev/
 
+.. _building.deps:
+
+Third-Party Dependencies
+########################
+
+When configuring and building |amongoc| with CMake, the configuration script
+will attempt to automatically obtain build-time dependencies using vcpkg_. This
+behavior can be disabled by setting :cmake:variable:`AMONGOC_USE_PMM` to
+``OFF``. Relying on vcpkg_ will attempt to build the third-party dependencies
+during project configuration, which may require additional configure-time
+dependencies.
+
+The following external libraries are required when **building** |amongoc|. If
+not relying vcpkg_, then they will need to be installed and available before
+configuring the project:
+
+- OpenSSL
+- Boost.URL and Boost.Container
+- `{fmt}`_
+- **If using vcpkg to install dependencies** (the default), then you will also
+  need to have Tar, cURL, Perl, Zip/Unzip, pkg-config, Make, and Bash. On Linux,
+  you will also need the linux development headers.
+
+The following packages are downloaded using CMake's FetchContent, and need not
+be manually installed:
+
+- Catch2_ - Only required for testing and only imported if
+  :cmake:variable:`BUILD_TESTING` is enabled.
+- Asio_ - Used for asynchronous I/O (Build-time only) (Note: Not ``Boost.Asio``!)
+- neo-fun_ (Build-time only)
```

Comment:
Is neo-fun also needed by consumers? I got a linker error building a test app using an installed `amongoc`:

```
/usr/bin/ld: /home/admin/c-bootstrap/amongoc/install/mongo-c-driver-async-pr2/lib/libamongoc.a(connection_pool.cpp.o): in function `neo::repr_detail::operator<<(std::ostream&, neo::repr_detail::item_repr_base const&)':
connection_pool.cpp:(.text._ZN3neo11repr_detaillsERSoRKNS0_14item_repr_baseE[_ZN3neo11repr_detaillsERSoRKNS0_14item_repr_baseE]+0x1f): undefined reference to `neo::repr_detail::item_repr_base::write_ostream(std::ostream&) const'
[...]
```

amongoc.a was installed following [these steps](https://github.com/kevinAlbs/c-bootstrap/tree/8b575f2a9cb23fd725e8ac92767066a7d55214b3/amongoc/investigations/test). Linking to neo-fun in the test app fixed the error:

```cmake
# Work around linker error:
target_link_libraries(hello-amongoc PRIVATE /home/admin/review/mongo-c-driver-async-2/_build/vcpkg_installed/x64-linux/lib/libneo-fun.a)
```